src/pages/workspaces/workspace/Data.js

```diff
@@ -419,38 +418,175 @@ class WorkspaceDataContent extends Component {
+                    onChange: e => this.setState({ editKey: e.target.value })
+                  }) :
+                  renderDataCell(filteredAttributes[rowIndex][0], namespace)
               },
               {
                 size: { grow: 1 },
                 headerRenderer: () => h(HeaderCell, ['Value']),
-                cellRenderer: ({ rowIndex }) => renderDataCell(filteredAttributes[rowIndex][1], namespace)
+                cellRenderer: ({ rowIndex }) => {
+                  const originalKey = filteredAttributes[rowIndex][0]
+                  const originalValue = filteredAttributes[rowIndex][1]
+
+                  return h(Fragment, [
+                    div({ style: { flex: 1, minWidth: 0, display: 'flex' } }, [
+                      editIndex === rowIndex ?
+                        textInput({
+                          value: editValue,
+                          onChange: e => this.setState({ editValue: e.target.value })
+                        }) :
+                        renderDataCell(originalValue, namespace)
+                    ]),
+                    editIndex === rowIndex ?
+                      h(Fragment, [
+                        h(Select, {
+                          wrapperStyle: { marginLeft: '1rem', width: 150 },
+                          autoFocus: false,
+                          searchable: false,
+                          clearable: false,
+                          value: editType,
+                          onChange: ({ value }) => this.setState({ editType: value }),
+                          options: _.map(name => {
+                            return { label: _.startCase(name), value: name }
+                          }, ['string', 'number', 'boolean', 'string list', 'number list', 'boolean list'])
+                        }),
+                        linkButton({
+                          tooltip: Utils.summarizeErrors(inputErrors) || 'Save changes',
+                          disabled: !!inputErrors.length,
+                          style: { marginLeft: '1rem' },
+                          onClick: async () => {
+                            const isList = editType.includes('list')
+                            const newBaseType = isList ? editType.slice(0, -5) : editType
+
+                            const parsedValue = isList ? _.map(Utils.convertValue(newBaseType), editValue.split(/,s*/)) :
+                              Utils.convertValue(newBaseType, editValue)
+
+                            this.setState({ loading: true })
+
+                            await Workspaces.workspace(namespace, name).shallowMergeNewAttributes({ [editKey]: parsedValue })
+
+                            if (editKey !== originalKey) {
+                              await Workspaces.workspace(namespace, name).deleteAttributes([originalKey])
+                            }
+
+                            await this.loadData()
+
+                            stopEditing()
+                          }
```

Comment:
TOL: This handler is very long and visually breaks up the component flow quite a bit. From the perspective of a reader, it might be nice to break this out into its own method.